crates/uv/tests/it/python_install.rs

```diff
@@ -3721,3 +3722,119 @@ fn python_install_build_version_pypy() {
     error: No download found for request: pypy-3.10-[PLATFORM]
     ");
 }
+
+#[test]
+fn python_install_upgrade() {
+    let context: TestContext = TestContext::new_with_versions(&[])
+        .with_python_download_cache()
+        .with_filtered_python_keys()
+        .with_filtered_exe_suffix()
+        .with_managed_python_dirs();
+
+    // Provide `--upgrade` as an `install` option without any versions
+    uv_snapshot!(context.filters(), context.python_install().arg("--upgrade"), @r"
+    success: true
+    exit_code: 0
+    ----- stdout -----
+
+    ----- stderr -----
+    Installed Python 3.14.0 in [TIME]
+     + cpython-3.14.0-[PLATFORM] (python3.14)
+    ");
```

Comment:
Another quirk: The output log for plain without any pin is the same with and without upgrade. I'd have expected something like "latest patch version Python 3.14.0 is already installed".

```
$ uv python install
Python is already installed. Use `uv python install <request>` to install another version.
$ uv python install --upgrade
Python is already installed. Use `uv python install <request>` to install another version.
```

Similar to what an explicit request tells you:

```
$ uv python install --upgrade 3.14
Python 3.14 is already on the latest supported patch release
```

It's pre-existing, but it would also be nice if `uv python install` and `uv python install --upgrade 3.14` would tell me the version that is installed that they consider latest.

Reply:
I think I might expect this to error asking for a request? but you're right the message should change if we don't.